java/datastructures/lib/src/main/java/datastructures/linkedlist/BinaryTree.java

```diff
@@ -63,4 +63,68 @@ public Object[] postOrder(TreeNode<T> node){
     return outputValues;
 
   }
+
+  public void add(T value) {
+    TreeNode<T> newNode = new TreeNode<>(value);
+
+    if (root == null) {
+      root = newNode;
+      return;
+    }
+
+    Queue<TreeNode<T>> queue = new Queue<>();
+    queue.enqueue(root);
+
+    while (!queue.isEmpty()) {
+      TreeNode<T> node = queue.dequeue();
+
+      if (node.leftNode == null) {
+        node.leftNode = newNode;
+        return;
+      } else {
+        queue.enqueue(node.leftNode);
+      }
+
+      if (node.rightNode == null) {
+        node.rightNode = newNode;
+        return;
+      } else {
+        queue.enqueue(node.rightNode);
+      }
+    }
+  }
+
+  public Integer findMaxFromBinaryTree(){
+    if (root == null){
+      return null;
+    }
+
+    Queue<TreeNode> breadth = new Queue();
```

Comment:
You can nest parameters - try writing this line as:

`Queue<TreeNode<T>> breadth = new Queue<>();`